### DISCUSSION TITLE
Add option to pause connection for short periods of time?

### DISCUSSION THREAD

**Discussion by audacenoire** [mullvad/mullvadvpn-app#8909] [Feature requests & Ideas]
I was wondering if it would be possible to add a feature to temporarily disconnect the VPN for a few minutes and have it reconnect automatically. 

Sometimes I need to turn my VPN off to access certain websites that block VPN access, but I keep forgetting to turn it back on. It would be extremely helpful if I could just pause the connection long enough to use the website and then have it come back on without me having to remember to do so.

**Comment by MrChocolatine** [comment]
I like this suggestion, it happens to me as well sometimes until I take a look at the tray icon.

Just another idea, to add to what @audacenoire already suggested, what about notifications like?

> Mullvad VPN is still disconnected, do you want to reconnect it?

That notification would:
- Show up every X minutes (fixed value? Customisation in the application's setting?)
- Have 2 buttons: `Reconnect` and `Keep disconnected`

We could also have the choice to enable/disable these notifications in the application's settings?

> Connection reminder `on/off`

**Comment by faern** [comment]
A safer way to fix the issue you are having is to have a separate browser that you exclude (split tunneling), and you launch that browser occasionally to load these misbehaving websites. If split tunneling is available on your platform that is. We have not implemented it on macOS or iOS yet.

In general we try to avoid any type of workaround that are timing based or "somewhat secure, but it depends on the timing of other things". During this disconnected period anything on your computer can reach the internet from your normal IP, not just the misbehaving website you are trying to load. It's of course always up to the threat model of each individual user. But we don't think these type of solutions are very clean.

**Comment by techwoes** [comment]
I 100% support this as it is a "good enough" option for the vast majority of people. 1/3/5/10 minute option would be great for logging into bank apps and such.

**Comment by nofeet** [comment]
For comparison, the PIA client has a pause timer that they call Snooze: https://helpdesk.privateinternetaccess.com/kb/articles/understanding-the-snooze-feature

**Comment by audacenoire** [comment]
Also split tunnelling doesn't always work, some apps and websites seem to detect the presence of the VPN proxy and continue to block it even when split tunneling is active and the item has been added to the whitelist 

This is across VPNs, it's not just a Mullvad thing.

**Comment by ghost** [comment]
FWIW, I too think this would be a nice feature. As of late, I have experienced more and more services blocking me when using Mullvad. This is of course not a Mullvad issue, but I tend to forget to turn the VPN back on. It would be great if it reconnected automatically after a given interval.

**Comment by alex-eliot** [comment]
> A safer way to fix the issue you are having is to have a separate browser that you exclude (split tunneling), and you launch that browser occasionally to load these misbehaving websites. If split tunneling is available on your platform that is. We have not implemented it on macOS or iOS yet.
> 
> In general we try to avoid any type of workaround that are timing based or "somewhat secure, but it depends on the timing of other things". During this disconnected period anything on your computer can reach the internet from your normal IP, not just the misbehaving website you are trying to load. It's of course always up to the threat model of each individual user. But we don't think these type of solutions are very clean.

Just to add my two cents on this: since it would be an action taken explicitly by the user, I don't think the security concern is very relevant. Users are already manually turning it off and then on, so it's a convenience to have the option of this kind of automation. More and more sites are blocking VPNs and server IPs and it would be extremely helpful to not have to always check whether it's enabled or not.

**Comment by MrChocolatine** [comment]
> More and more sites are blocking VPNs and server IPs

Just a personal opinion but I think it got even worse in the last 6 months.
It's very subjective I know.

**Comment by dragomang87** [comment]
Just to get things started and get an idea of how much work it would be, where would such a timer for the auto-reconnect go internally? I don't recall there being any timed state in the app, it's all timeless, so is it better to implement internally or would it be better to use the OS? (e.g. trigger some inhibitors for systemd/Linux)

**Reply by faern** [reply]
If implemented in the OS, then every platform would need their own solution, making the total complexity much higher compared to if it was integrated into the `mullvad-daemon` and similar components.

If this was to be implemented, I think it would be some kind of timer task living in the `mullvad-daemon` or adjacent crates. We have many timed recurring tasks already. Such as fetching an updated server list every hour. In a similar fashion this could be implemented as a timer that starts on user request, and triggers a connect event when it finishes. And of course aborts if certain other actions are performed in between.

**Comment by Buddel1** [comment]
I like @audacenoire 's idea. I would also like to be able to pause my VPN connection for a couple of minutes/hours without having to terminate VPN connection.

**Comment by benjamin-sousa** [comment]
I used iOS Shortcuts to toggle VPN, wait 5 minutes then toggle it again. Suggesting it here for those looking for an immediate workaround, but I agree it would be a convenient feature in Mullvad.
